runtime/parachains/src/assigner_on_demand/tests.rs

```diff
@@ -0,0 +1,276 @@
+
+		// In the real runtime this is expected to be called by the `InclusionInherent` pallet.
+		Scheduler::update_claimqueue(BTreeMap::new(), b + 1);
+	}
+}
+
+#[derive(Debug)]
+pub(super) struct GenesisConfigBuilder {
+	parathread_cores: u32,
+	on_demand_base_fee: Balance,
+	on_demand_fee_variability: Perbill,
+	on_demand_max_queue_size: u32,
+	on_demand_target_queue_utilization: Perbill,
+}
+
+impl Default for GenesisConfigBuilder {
+	fn default() -> Self {
+		Self {
+			parathread_cores: 10,
+			on_demand_base_fee: 10_000,
+			on_demand_fee_variability: Perbill::from_percent(1),
+			on_demand_max_queue_size: 100,
+			on_demand_target_queue_utilization: Perbill::from_percent(25),
+		}
+	}
+}
+
+impl GenesisConfigBuilder {
+	pub(super) fn build(self) -> crate::mock::MockGenesisConfig {
+		let mut genesis = default_genesis_config();
+		let config = &mut genesis.configuration.config;
+		config.parathread_cores = self.parathread_cores;
+		config.on_demand_base_fee = self.on_demand_base_fee;
+		config.on_demand_fee_variability = self.on_demand_fee_variability;
+		config.on_demand_queue_max_size = self.on_demand_max_queue_size;
+		config.on_demand_target_queue_utilization = self.on_demand_target_queue_utilization;
+		genesis
+	}
+}
+
+#[test]
+fn spot_traffic_capacity_zero_returns_none() {
+	let res = OnDemandAssigner::calculate_spot_traffic(
+		FixedU128::from(u128::MAX),
+		0u32,
+		u32::MAX,
+		Perbill::from_percent(100),
+		Perbill::from_percent(1),
+	);
+	assert_eq!(res, None)
+}
+
+#[test]
+fn spot_traffic_queue_size_larger_than_capacity_returns_none() {
+	let res = OnDemandAssigner::calculate_spot_traffic(
+		FixedU128::from(u128::MAX),
+		1u32,
+		2u32,
+		Perbill::from_percent(100),
+		Perbill::from_percent(1),
+	);
+	assert_eq!(res, None)
+}
+
+#[test]
+fn spot_traffic_calculation_identity() {
+	let res = OnDemandAssigner::calculate_spot_traffic(
+		FixedU128::from_u32(1),
+		1000,
+		100,
+		Perbill::from_percent(10),
+		Perbill::from_percent(3),
+	);
+	assert_eq!(res.unwrap(), FixedU128::from_u32(1))
+}
+
+#[test]
+fn spot_traffic_calculation_u32_max() {
+	let res = OnDemandAssigner::calculate_spot_traffic(
+		FixedU128::from_u32(1),
+		u32::MAX,
+		u32::MAX,
+		Perbill::from_percent(100),
+		Perbill::from_percent(3),
+	);
+	assert_eq!(res.unwrap(), FixedU128::from_u32(1))
+}
+
+#[test]
+fn spot_traffic_calculation_u32_traffic_max() {
+	let res = OnDemandAssigner::calculate_spot_traffic(
+		FixedU128::from(u128::MAX),
+		u32::MAX,
+		u32::MAX,
+		Perbill::from_percent(1),
+		Perbill::from_percent(1),
+	);
+	assert_eq!(res, None)
+}
+
+#[test]
+fn sustained_target_increases_spot_traffic() {
+	let mut traffic = FixedU128::from_u32(1u32);
+	for _ in 0..50 {
+		traffic = OnDemandAssigner::calculate_spot_traffic(
+			traffic,
+			100,
+			12,
+			Perbill::from_percent(10),
+			Perbill::from_percent(100),
+		)
+		.unwrap()
+	}
+	assert_eq!(traffic, FixedU128::from_inner(2_718_103_312_071_174_051u128))
+}
+
+#[test]
+fn place_order_works() {
```

Comment:
Hmm, this test seems to be mostly testing that an order does not work (but misses a few reasons).

Reply:
Test now does what it says on the tin, albeit with checks along the way.